roles/content_view_version_cleanup/tasks/delete_cv_versions.yml

```diff
@@ -0,0 +1,14 @@
+---
+- name: "delete content view versions of {{ cv_name }}"
+  theforeman.foreman.content_view_version:
+    server_url: "{{ server_url | default(omit) }}"
+    username: "{{ username | default(omit) }}"
+    password: "{{ password | default(omit) }}"
+    validate_certs: "{{ validate_certs | default(omit) }}"
+    organization: "{{ organization }}"
```

Comment:
I'd like to consider introducing a role for common variables that is a dependency of other roles so as to define the variables in a common place. Naming is hard for such a role:

server_variables
api_variales
common_variables
?!?!!

Reply:
so that you don't have to define the same vars every time? good idea, but I fear that'd clash with the environment fallback?

Reply:
What is 'environment fallback' ?

Reply:
Since 1.3.0 users can set `FOREMAN_USERNAME` and similar as environment variables and Ansible will pick that up instead of passing params to the modules

Reply:
Ansible at what layer picks them up?  Is that our convention or Ansible's?

Reply:
Ansible's. See https://github.com/theforeman/foreman-ansible-modules/issues/837 and linked PR ;)

Reply:
Also i don't see, why we should put them in a common role. What would that role contain if we cannot even provide meaningful defaults for them. I am however in favor of using the same variable names in all roles. Then you'd provide them once in the playbook, as module_defaults or via environment. And if you need different sets of server vars in two roles, you can provide them at `include_role`.
The environment is the last resort. If is not provided as a value the module itself will fallback to using the environment.

Reply:
The value of a role would be moving common things into a single place. Define them in one place, set the role as a dependency and it's easier to follow the chain or know where the variable originates from rather than relying on convention to prevail. 

I'm having a hard time groking why environment variable fallback breaks if a single role defines a common set of variables.

Reply:
I'm not against having a common role to collect common things. But take `password`. If you define that in the common role, you need to assign a (default) value to it. What would that be?
Always having a value prevents us from using the `default(omit)` mechanism.
OTOH nothing stops us from expecting the same variable `foreman_password` in all the roles with the env-fallback without it being predefined. Maybe the common role could document that behavior.